exec/java-exec/src/main/java/org/apache/drill/exec/store/log/README.md

```diff
@@ -143,6 +143,14 @@ cardinality.
 You may find it helpful to specify the regex and column names via the plugin
 config, types via the `CREATE SCHEMA` command.
 
+## Table Functions
+
+Log files come in many forms. It would be very convenient to use Drill table
```

Comment:
Table function will work for all log format properties, except of list. Knowing that list is not supported, does it makes sense to replace list schema parameter with String and rename it to avoid clash with schema parameter for schema provisioning.

Reply:
As I recall, Drill does not have a good way to deal with changes to the schema of a storage plugin. Some time back, I remember struggling to understand why my server would not start, only to eventually learn that some plugin or other changed its config and so Drill failed when trying to load the existing config from ZK. Has this been fixed?

If we change schema to a string, we'd need to run code to convert old configs. Also, we'd have the problem of what to do with the type property. We could not easily convert an existing config into a table schema.

Given these uncertainties, my thought was to leave the config alone and try to fit in the provided schema as best we can on top of the existing config.

Can you think of a better approach?

Reply:
I guess initial choice of list property did not take into account that it does not work with table function. I don't think you can fix this backward compatibility in ZK but since this plugin is a role model for others I think it should have proper configuration, so changing `schema` property to be `String` instead `List` might be reasonable. Or can we have both properties one in String, another in `List<String>`? We can indicate in release notes that log plugin has been changed and config in ZK must be updated.